hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataStorage.java

```diff
@@ -224,7 +234,12 @@ synchronized void recoverTransitionRead(DataNode datanode,
     try {
       for (int idx = 0; idx < getNumStorageDirs(); idx++) {
         doTransition(datanode, getStorageDir(idx), nsInfo, startOpt);
-        createStorageID(getStorageDir(idx));
+        boolean haveValidStorageId =
```

Comment:
I've had to add this by myself because in branch-2.7.1, the createStorageID on line 242 has been removed.